core/src/test/scala/unit/kafka/server/LeaveGroupRequestTest.scala

```diff
@@ -0,0 +1,141 @@
+    new ClusterConfigProperty(key = "group.coordinator.new.enable", value = "false"),
+    new ClusterConfigProperty(key = "offsets.topic.num.partitions", value = "1"),
+    new ClusterConfigProperty(key = "offsets.topic.replication.factor", value = "1")
+  ))
+  def testLeaveGroupWithOldConsumerGroupProtocolAndOldGroupCoordinator(): Unit = {
+    testLeaveGroup()
+  }
+
+  private def testLeaveGroup(): Unit = {
+
+    // Creates the __consumer_offsets topics because it won't be created automatically
+    // in this test because it does not use FindCoordinator API.
+    createOffsetsTopic()
+
+    // Create the topic.
+    createTopic(
+      topic = "foo",
+      numPartitions = 3
+    )
+
+    for (version <- ApiKeys.LEAVE_GROUP.oldestVersion() to ApiKeys.LEAVE_GROUP.latestVersion(isUnstableApiEnabled)) {
+
+      // Join the consumer group. Note that we don't heartbeat here so we must use
+      // a session long enough for the duration of the test.
+      val (memberId1, _) = joinDynamicConsumerGroupWithOldProtocol("grp-1")
+      if (version >= 3) {
+        joinStaticConsumerGroupWithOldProtocol("grp-2", "group-instance-id")
+      }
+
+      // Request with empty group id.
+      leaveGroupWithOldProtocol(
+        groupId = "",
+        memberIds = List(memberId1),
+        expectedLeaveGroupError = Errors.INVALID_GROUP_ID,
+        expectedMemberErrors = List(Errors.NONE),
+        version = version.toShort
+      )
+
+      // Request with invalid group id and unknown member id should still get Errors.INVALID_GROUP_ID.
+      leaveGroupWithOldProtocol(
+        groupId = "",
+        memberIds = List("member-id-unknown"),
+        expectedLeaveGroupError = Errors.INVALID_GROUP_ID,
+        expectedMemberErrors = List(Errors.NONE),
+        version = version.toShort
+      )
+
+      // Request with unknown group id gets Errors.UNKNOWN_MEMBER_ID.
+      leaveGroupWithOldProtocol(
+        groupId = "grp-unknown",
+        memberIds = List(memberId1),
+        expectedLeaveGroupError = if (version >= 3) Errors.NONE else Errors.UNKNOWN_MEMBER_ID,
+        expectedMemberErrors = if (version >= 3) List(Errors.UNKNOWN_MEMBER_ID) else List.empty,
+        version = version.toShort
+      )
+
+      // Request with unknown member ids.
+      leaveGroupWithOldProtocol(
+        groupId = "grp-1",
+        memberIds = if (version >= 3) List("unknown-member-id", JoinGroupRequest.UNKNOWN_MEMBER_ID) else List("unknown-member-id"),
+        expectedLeaveGroupError = if (version >= 3) Errors.NONE else Errors.UNKNOWN_MEMBER_ID,
+        expectedMemberErrors = if (version >= 3) List(Errors.UNKNOWN_MEMBER_ID, Errors.UNKNOWN_MEMBER_ID) else List.empty,
+        version = version.toShort
+      )
+
+      // Success GroupLeave request.
+      leaveGroupWithOldProtocol(
+        groupId = "grp-1",
+        memberIds = if (version >= 3) List(memberId1) else List(memberId1),
```

Comment:
I don't get this one. I suppose that we could just remove the version check, isn't it?